source/reference/ejson/deserialize.txt

```diff
@@ -0,0 +1,200 @@
+use:
+
+.. code-block:: javascript
+   :copyable: false
+
+   EJSON.deserialize( object, [ options ] )
+
+To run ``EJSON.deserialize()`` from the system command line, use:
+
+.. code-block:: shell
+   :copyable: false
+
+   mongosh --eval "EJSON.deserialize( object, [ options ] )"
+
+Examples
+--------
+
+Create the ``sales`` collection for the examples:
+
+.. code-block:: javascript
+
+   db.sales.insertMany( [ 
+      { custId: 345, purchaseDate: ISODate("2023-07-04"),
+        quantity: 4, cost: Decimal128("100.60") },
+      { custId: 346, purchaseDate: ISODate("2023-07-12"),
+        quantity: 3, cost: Decimal128("175.45") },
+      { custId: 486, purchaseDate: ISODate("2023-08-01"),
+        quantity: 9, cost: Decimal128("200.53") }
+   ] )
+
+Interactive Mongo Shell EJSON.deserialize() Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example retrieves the ``sales`` documents as an array and
+stores the results in the ``salesCollection`` object:
+
+.. code-block:: javascript
+
+   salesCollection = EJSON.deserialize( db.sales.find().toArray() )
+
+Example output:
+
+.. code-block:: javascript
+   :copyable: false
+
+   [
+      {
+         _id: '6520519a0dbd2d208a5c7941',
+         custId: 345,
+         purchaseDate: '2023-07-04T00:00:00.000Z',
+         quantity: 4,
+         cost: Decimal128("100.60")
+      },
+      {
+         _id: '6520519a0dbd2d208a5c7942',
+         custId: 346,
+         purchaseDate: '2023-07-12T00:00:00.000Z',
+         quantity: 3,
+         cost: Decimal128("175.45")
+      },
+      {
+         _id: '6520519a0dbd2d208a5c7943',
+         custId: 486,
+         purchaseDate: '2023-08-01T00:00:00.000Z',
+         quantity: 9,
+         cost: Decimal128("200.53")
+      }
+   ]
+
+Command Line Mongo Shell EJSON.deserialize() Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following sections show how to create an example file and then
+import the file with an ``EJSON.deserialize()`` example.
+
+Create the Example File
+```````````````````````
+
+The following example retrieves the ``sales`` documents as an array and
+saves the results to a file named ``salesDeserialize.json`` on the
+computer's file system:
+
+.. code-block:: javascript
+
+   let salesCollection = EJSON.stringify( db.sales.find().toArray() )   
+
+   fs.writeFileSync( 'salesDeserialize.json', salesCollection ) 
+
+Import the Example File from the Command Line
+`````````````````````````````````````````````
+
+To import the ``salesDeserialize.json`` file and create a new collection
+named ``salesFromDeserializeFile``, exit ``mongosh`` and then run this
+example from the command line:
+
+.. code-block:: sh
+
+   # Note: The example is formatted to fit the page.
+
+   mongosh --quiet  --eval "db.salesFromDeserializeFile.insertMany( \
+      EJSON.deserialize( JSON.parse ( \
+      fs.readFileSync( 'salesDeserialize.json', 'utf8' ) ) ) )"
```

Comment:
It’s probably worth mentioning that, for this specific example, `EJSON.parse()` would be the more common approach, and that manually splitting it up into `JSON.parse()` + `EJSON.deserialize()` just gives the user a bit more flexibility.